scipy/_lib/_array_api.py

```diff
@@ -155,18 +155,14 @@ def _asarray(
     """
     if xp is None:
         xp = array_namespace(array)
-    if xp.__name__ in {"numpy", "scipy._lib.array_api_compat.numpy"}:
+    if is_numpy(xp):
         # Use NumPy API to support order
         if copy is True:
             array = np.array(array, order=order, dtype=dtype, subok=subok)
         elif subok:
             array = np.asanyarray(array, order=order, dtype=dtype)
         else:
             array = np.asarray(array, order=order, dtype=dtype)
-
-        # At this point array is a NumPy ndarray. We convert it to an array
-        # container that is consistent with the input's namespace.
-        array = xp.asarray(array)
```

Comment:
Any guesses what this was for?

Reply:
historical cruft I believe - looks like it initially always used NumPy regardless of array type, then later this was guarded for only NumPy arrays (my guess)